src/messages/embed/Embed.ts

```diff
@@ -0,0 +1,304 @@
+
+		this.author = name ? { name, icon_url: iconURL ?? undefined, url: url ?? undefined } : undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the color of this embed.
+	 * @param color The color of the embed.
+	 */
+	public setColor(color: number | null): this {
+		// Data assertions
+		ow(color, 'color', colorPredicate);
+
+		this.color = color ?? undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the description of this embed.
+	 * @param description The description.
+	 */
+	public setDescription(description: string | null): this {
+		// Data assertions
+		ow(description, 'description', descriptionPredicate);
+
+		this.description = description ?? undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the footer of this embed.
+	 * @param text The text of the footer.
+	 * @param iconURL The icon URL of the footer.
+	 */
+	public setFooter(text: string, iconURL?: string | null): this {
+		// Data assertions
+		ow(text, 'text', footerTextPredicate);
+		ow(iconURL, 'iconURL', urlPredicate);
+
+		this.footer = text ? { text, icon_url: iconURL ?? undefined } : undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the image of this embed.
+	 * @param url The URL of the image.
+	 */
+	public setImage(url: string | null): this {
+		// Data assertions
+		ow(url, 'url', urlPredicate);
+
+		this.image = url ? { url } : undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the thumbnail of this embed.
+	 * @param url The URL of the thumbnail.
+	 */
+	public setThumbnail(url: string | null): this {
+		// Data assertions
+		ow(url, 'url', urlPredicate);
+
+		this.thumbnail = url ? { url } : undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the timestamp of this embed.
+	 * @param timestamp The timestamp or date.
+	 */
+	public setTimestamp(timestamp: number | Date | null = Date.now()): this {
+		// Data assertions
+		ow(timestamp, 'timestamp', timestampPredicate);
+
+		this.timestamp = timestamp ? new Date(timestamp).toISOString() : undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the title of this embed.
+	 * @param title The title.
+	 */
+	public setTitle(title: string | null): this {
+		// Data assertions
+		ow(title, 'title', titlePredicate);
+
+		this.title = title ?? undefined;
+		return this;
+	}
+
+	/**
+	 * Sets the URL of this embed.
+	 * @param url The URL.
+	 */
+	public setURL(url: string | null): this {
+		// Data assertions
+		ow(url, 'url', urlPredicate);
+
+		this.url = url ?? undefined;
+		return this;
+	}
+
+	/**
+	 * Transforms the embed to a plain object.
+	 */
+	public toJSON(): APIEmbed {
+		return { ...this };
+	}
+
+	/**
+	 * Normalizes field input and resolves strings.
+	 * @param fields Fields to normalize.
+	 */
+	public static normalizeFields(...fields: APIEmbedField[]): APIEmbedField[] {
+		return fields.flat(Infinity).map((field) => {
+			ow(field.name, fieldNamePredicate);
+			ow(field.value, fieldValuePredicate);
+			ow(field.inline, fieldInlinePredicate);
+
+			return { name: field.name, value: field.value, inline: field.inline ?? false };
```

Comment:
Should mark inline as undefined instead of false

```suggestion
			return { name: field.name, value: field.value, inline: field.inline ?? undefined };
```